src/module-elasticsuite-virtual-category/Plugin/Widget/ProductsListPlugin.php

```diff
@@ -136,15 +155,31 @@ public function afterCreateCollection(ProductsList $subject, $collection)
                 $conditions = $this->conditionsHelper->decode($conditions);
                 foreach ($conditions as $condition) {
                     if (!empty($condition['attribute'])) {
-                        if ($condition['attribute'] == 'category_ids') {
-                            if (array_key_exists('value', $condition)) {
-                                $categoryId = $condition['value'];
-                                try {
+                        if ($condition['attribute'] == 'category_ids' && array_key_exists('value', $condition)) {
```

Comment:
I think we could do something more simple.

We should be handle to handle $condition['value'] in the same manner, no matter if it contains "1" or "1,2", the explode will always return an array, no ?

I have something like this in my mind : 

```php
  if ($condition['attribute'] == 'category_ids' && array_key_exists('value', $condition)) {
          $filterQueries = [];
          
          // This should work even when $condition['value'] contains only one value, no ?

          $categoryIds = array_map("trim", explode(',', (string) $condition['value']));
          foreach ($categoryIds as $categoryId) {
              try {
                  $category = $this->categoryRepository->get($categoryId, $storeId);
                  $filterQueries[] = $this->filterProvider->getQueryFilter($category);
              } catch (NoSuchEntityException $exception) {
                  continue;
              }
          }

          if (!empty($filterQueries) {
              $query = $this->queryFactory->create(QueryInterface::TYPE_BOOL, ['should' => $filterQueries]);
              $collection->addQueryFilter($query);
          }
          continue;
      }
  }
```

Reply:
Yes, you're right. I refactored the code.